src/plugins/home/public/application/components/guided_onboarding/getting_started.tsx

```diff
@@ -48,7 +49,12 @@ export const GettingStarted = () => {
   const [guidesState, setGuidesState] = useState<GuideState[]>([]);
   const [isLoading, setIsLoading] = useState<boolean>(false);
   const [isError, setIsError] = useState<boolean>(false);
-  const [filter, setFilter] = useState<GuideFilterValues>('all');
+  const { search } = useLocation();
+  const query = parse(search);
+
+  const [filter, setFilter] = useState<GuideFilterValues>(
+    query.cloudDiscoveryUseCase ? (query.cloudDiscoveryUseCase as GuideFilterValues) : 'all'
```

Comment:
[nit] I might make the query param more generic, e.g., "useCase" since it is visible to the user